### ISSUE TITLE
Add `aria-owns` in `MenuProvider` to define ownership relationship for accessibility

### ISSUE THREAD

**Issue by spli02** [fortanix/baklava#405]
The current `MenuProvider` does not define an explicit relationship between the anchor element and the menu.
`aria-owns` allows an element to reference elements that are not its DOM children, but should be treated as part of its accessibility tree.
To improve accessibility for screen readers, it's better use the `aria-owns` attribute to link the anchor to the menu element.

Ref: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-owns
https://www.digitala11y.com/aria-owns-properties/
